javascript/node/selenium-webdriver/safari.js

```diff
@@ -200,14 +217,19 @@ class Driver extends webdriver.WebDriver {
    * @return {!Driver} A new driver instance.
    */
   static createSession(opt_config, opt_flow) {
-    let caps;
+    let caps, exe;
+
     if (opt_config instanceof Options) {
       caps = opt_config.toCapabilities();
     } else {
-      caps = opt_config || Capabilities.safari()
+      caps = opt_config || Capabilities.safari();
+    }
+
+    if (caps.get(TECHNOLOGY_PREVIEW_CAPABILITY_KEY)) {
```

Comment:
I meant you need to properly handle extracting options from the capabilities object.
For consistency with Java, you should end up with something alone the lines of

```js
// Some details omitted for brevity

class Options {
  setTechnologyPreview(preview) {
    this.options_['technologyPreview'] = !!preview;
    return this;
  }
}

function useTechnologyPreview(obj) {
    if (obj instanceof Options) {
      return !!obj.options_['technologyPreview']
    }

    if (obj && typeof obj === 'object') {
      return !!obj['technologyPreview'];
    }

    return false;
}


class Driver extends webdriver.WebDriver {
  static createSession(opt_config, opt_flow) {
    let caps, exe;

    if (opt_config instanceof Options) {
      caps = opt_config.toCapabilities();

    } else {
      caps = opt_config || Capabilities.safari();
    }

    if (useTechnologyPreview(caps.get(OPTIONS_CAPABILITY_KEY))) {
      exe = SAFARIDRIVER_TECHNOLOGY_PREVIEW_EXE;
    }
    // ...
  }
}
```